src/utils/base32.js

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (C) 2011 by Isaac Wolkerstorfer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+*/
+// This implementation comes from the project
+// https://github.com/agnoster/base32-js, from this revision:
```

Comment:
This was the library linked from the wikipedia article. I took it because it was self-contained and and the alphabet was easy to change. I slightly audited the code too.

Reply:
I imported it so that:
* I can change the alphabet
* it's easier to add the types ;)